src/hs_checkpoint_reader.c

```diff
@@ -257,3 +258,64 @@ void hs_output_checkpoints(hs_checkpoint_reader *cpr, FILE *fh)
   }
   pthread_mutex_unlock(&cpr->lock);
 }
+
+
+void hs_remove_checkpoint(hs_checkpoint_reader *cpr,
+                          const char *key)
+{
+  pthread_mutex_lock(&cpr->lock);
+  lua_pushstring(cpr->values, key);
+  lua_pushnil(cpr->values);
+  lua_settable(cpr->values, LUA_GLOBALSINDEX);
```

Comment:
I've factorized common code in an unsafe (does not acquire mutex lock) static function in 0ee9f9ad61974c6ee98fc41f586345cfabb179c0 : is it what you were thinking about?
I'm also now using `lua_setfield` instead of `lua_settable`.